### PR TITLE
gem5 fixes and one AndroidDevice fix

### PR DESCRIPTION
**Pull request by Sticklyman1936**
gem5:
 * Removed busybox dependency for gem5 devices
 * Moved resize shell commands to own method
 * Fixed rare shell connection issues by trying up to 10 times to connect to the simulated system's shell
 * Added the m5 binary to Workload Automation. This is now deployed to the device automatically. The m5 binary is now a partial dependency as it is required for checkpoint support.
 * No longer wait for disabled boot animation
 * Allow the user to specify a login prompt and a corresponding password prompt from the agenda.
 * Removed unused methods
 * Remove the rename in pull_file to align with gem5

Android:
 * Use content instead of settings to get ANDROID_ID

### CODE REVIEW

File: wlauto/common/gem5/device.py [hunk 1]
```diff
@@ -384,8 +393,11 @@ def push_file(self, source, dest, **kwargs):
 
         # Back to the gem5 world
         self.gem5_shell("ls -al /mnt/obb/{}".format(filename))
-        self.gem5_shell("busybox cp /mnt/obb/{} {}".format(filename, dest))
-        self.gem5_shell("busybox sync")
+        if self.busybox:
+            self.gem5_shell("busybox cp /mnt/obb/{} {}".format(filename, dest))
```
Comment (setrofim) [inline wlauto/common/gem5/device.py]: should use `self.busybox` as above.

File: wlauto/common/gem5/device.py [hunk 2]
```diff
@@ -618,6 +631,19 @@ def sync_gem5_shell(self):
...
 
+    def resize_shell(self):
+        """
+        Resize the shell to avoid line wrapping issues.
+
+        """
+        # Try and avoid line wrapping as much as possible. Don't check the error
+        # codes from these command because some of them WILL fail.
+        self.gem5_shell('stty columns 1024', check_exit_code=False)
+        self.gem5_shell('busybox stty columns 1024', check_exit_code=False)
```
Comment (setrofim) [inline wlauto/common/gem5/device.py]: `self.busybox`

File: wlauto/common/gem5/device.py [hunk 3]
```diff
@@ -618,6 +631,19 @@ def sync_gem5_shell(self):
...
+        self.gem5_shell('stty cols 1024', check_exit_code=False)
+        self.gem5_shell('busybox stty cols 1024', check_exit_code=False)
```
Comment (setrofim) [inline wlauto/common/gem5/device.py]: `self.busybox`

File: wlauto/devices/android/gem5/__init__.py [hunk 4]
```diff
@@ -146,8 +156,11 @@ def install_executable(self, filepath, with_name=None):
         on_device_file = self.path.join(self.working_directory, executable_name)
         on_device_executable = self.path.join(self.binaries_directory, executable_name)
         self.push_file(filepath, on_device_file)
-        self.execute('busybox cp {} {}'.format(on_device_file, on_device_executable))
-        self.execute('busybox chmod 0777 {}'.format(on_device_executable))
+        if self.busybox:
+            self.execute('busybox cp {} {}'.format(on_device_file, on_device_executable))
+        else:
```
Comment (setrofim) [inline wlauto/devices/android/gem5/__init__.py]: `self.busybox`

**Comment by Sticklyman1936** [comment]
I have addressed the above issues and have added two new patches for you to review. The new patches are:

 * Gem5Device: Removed unused methods	d635596
 * Gem5Device: Remove the rename in pull_file to align with gem5	72a617c

**Pull request merged**